gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/tasks/CreateManifestTaskTest.groovy

```diff
@@ -1,10 +1,156 @@
+import com.palantir.gradle.dist.GradleTestSpec
+import nebula.test.dependencies.DependencyGraph
+import nebula.test.dependencies.GradleDependencyGenerator
 import org.gradle.api.Project
 import org.gradle.testfixtures.ProjectBuilder
-import spock.lang.Specification
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
 
-class CreateManifestTaskTest extends Specification {
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+
+class CreateManifestTaskTest extends GradleTestSpec {
+
+    def jsonMapper = new ObjectMapper()
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .setPropertyNamingStrategy(new KebabCaseStrategy())
+
+    File mavenRepo
+
+    def 'Fail on missing recommended product dependencies'() {
+        setup:
+        generateDependencies()
+        buildFile << """
+            plugins {
+                id 'com.palantir.sls-java-service-distribution'
+            }
+
+            repositories {
+                maven {url "file:///${mavenRepo.getAbsolutePath()}"}
+            }
+
+            project.version = '1.0.0'
+
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            task testCreateManifest(type: com.palantir.gradle.dist.tasks.CreateManifestTask) {
+                serviceName = "serviceName"
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = []
+                productDependenciesConfig = configurations.runtime
+            }
+        """.stripIndent()
+
+        when:
+        BuildResult buildResult = run(':testCreateManifest').buildAndFail()
+
+        then:
+        buildResult.task(':testCreateManifest').outcome == TaskOutcome.FAILED
+        buildResult.output.contains("The following products are recommended as dependencies but do not appear in the " +
+                "product dependencies or product dependencies ignored list: [group:name2, group:name]")
+    }
+
+    def 'Can ignore recommended product dependencies'() {
+        setup:
+        generateDependencies()
+        buildFile << """
+            plugins {
+                id 'com.palantir.sls-java-service-distribution'
+            }
+
+            repositories {
+                maven {url "file:///${mavenRepo.getAbsolutePath()}"}
+            }
+
+            project.version = '1.0.0'
+
+            dependencies {
+                runtime 'a:a:1.0'
+            }
+
+            task testCreateManifest(type: com.palantir.gradle.dist.tasks.CreateManifestTask) {
+                serviceName = "serviceName"
+                serviceGroup = "serviceGroup"
+                productType = "service"
+                manifestExtensions = [:]
+                productDependencies = []
+                productDependenciesConfig = configurations.runtime
+                ignoredProductIds = [
```

Comment:
ah - cause this test was testing just the task separately (i.e. this test creates it's own task "testCreateManifest" instead of using the conventional "createManifest")... I could change all the tests do use "createManifest" and then invoke the task and use "-x downStreamTask1 -x downStreamTask2, etc", but decided against cause im lazy >.<